### ISSUE TITLE
Exception Handling for Missing Data

### ISSUE THREAD

**Issue by reachtarunhere** [thugs-rumal/rumal#18]
In cases when the GeoIP plugin is disabled and the scans are run it results in JS errors because of missing exception handling for the given data.

The required changes need to be made in real_setPanels.js and  real_setMap.js  both of which are present in  rumal/templates/static/js/ 

If anyone is willing to take on this bug please comment below.

**Comment by sevdog** [comment]
It would be fun to fix this. I just need to re-setup the enviroment for this project and read the new docs.

I will update this issue when I have news.

**Comment by Dennisparchkov** [comment]
Great. Let us know if there are issues with setting up the environment, specifically for the back-end because Thug has had a recent modification (changing the way it is called).

**Comment by sevdog** [comment]
Even using the dockerized version there could be any problem with thug?

I was hoping it would work well. However if I get any problem I will tell you.

**Comment by Dennisparchkov** [comment]
Yes it seems there is a problem with the dockerized version when running `docker-compose build`, with installing mongodb on the back-end (https://github.com/thugs-rumal/rumal_docker/issues/13)

**Comment by sevdog** [comment]
Ok, I have reproduced the issue (after a long fight with docker and mongodb). The error seems to be this:
```
Uncaught TypeError: Cannot read property 'city' of undefinedloadMapPoints 
@ real_setMap.js:66(anonymous function) 
@ real_setMap.js:26(anonymous function) 
@ d3.min.js:1t 
@ d3.min.js:1u 
@ d3.min.js:1
```

As soon as I got some other time I wil try to figure out how to avoid this.

**Comment by sevdog** [comment]
I think this should solve the error issue:
```
@@ -63,10 +63,12 @@ function setMap(tooltip) {
 function loadMapPoints() {
     var points = [];
     for (var i in full_analysis["flat_tree"]) {
-        if ("City" in full_analysis["flat_tree"][i]["GeoPlugin"]["city"]) {
-            var city = full_analysis["flat_tree"][i]["GeoPlugin"]["city"];
+        var iflat_tree = full_analysis['flat_tree'][i];
+        if ('GeoPlugin' in iflat_tree && 'city' in iflat_tree['GeoPlugin'] && "City" in iflat_tree["GeoPlugin"]["city"]) {
+            var city = iflat_tree["GeoPlugin"]["city"];
             points.push(city);
         }
     }
     return points;
}
```

The double check for the presence of data should avoid the error. 

However, before committing, I have noticed that in this configuration the map is not displayed, it is only displayed a circle:
![image](https://cloud.githubusercontent.com/assets/13779643/20490064/f86f48be-b00c-11e6-8b6b-707f4d09b687.png)

but I cannot see any JS error, will check what is going on.

**Comment by Dennisparchkov** [comment]
The validation of the GeoPlugin looks good, i think it should work.

To change to the map (from your screenshot), click on the blue icon with 2 arrows <- -> located on the bottom right (This should display an empty map).
But, to test if this will work with the working GeoPlugin you must first configure it ([link to configuration](http://thugs-rumal.github.io/advancedusage.html)).

You can have a look at the docker front-end .sh file for how it should be setup [(docker front-end sh file)](https://github.com/thugs-rumal/rumal_docker/blob/master/frontend/start.sh#L34-L48)

**Comment by sevdog** [comment]
Ok, btw quite strange behaviour.

I'm going to test if everything goes fine with GeoPlugin, then I will inform you.

**Comment by sevdog** [comment]
GeoPlugin seems to work fine, going to commit and making PR.

**Comment by Dennisparchkov** [comment]
Great

**Comment by Dennisparchkov** [comment]
Fixed in #49
